genesis-programs/src/lib.rs

```diff
@@ -27,10 +27,21 @@ pub fn get_inflation(operating_mode: OperatingMode, epoch: Epoch) -> Option<Infl
 }
 
 pub fn get_inflation_for_epoch(operating_mode: OperatingMode, epoch: Epoch) -> Inflation {
+    // use nearly inpractical value while avoiding false positive clippy::absurd_extreme_comparisons
+    const DISTANT_FUTURE_EPOCH: Epoch = std::u64::MAX - 1;
+
     match operating_mode {
         OperatingMode::Development => Inflation::default(),
-        OperatingMode::Stable | OperatingMode::Preview => {
-            if epoch == std::u64::MAX {
+        OperatingMode::Preview => {
+            // Completely disable inflation prior to ship the inflation fix.
+            if epoch >= 68 {
```

Comment:
changed from `==` to `>=`. Otherwise, when we enter epoch 69, we're back to `::default()` Inflation. I think this is pretty dangerous. ;)

Reply:
This is because the caller expects level-triggering-like signalling, instead of edge-triggering-like signalling from this `fn`.

Reply:
yeah!
`get_inflation_for_epoch()` is now dead code on `master`, so we could revert back to the original version of `get_inflation()` that didn't have this major footgun: https://github.com/solana-labs/solana/blob/d8e885f4259e6c7db420cce513cb34ebf961073d/genesis-programs/src/lib.rs

Reply:
I reverted!